pandas/core/dtypes/dtypes.py

```diff
@@ -534,17 +541,24 @@ def update_dtype(self, dtype: 'CategoricalDtype') -> 'CategoricalDtype':
             msg = ('a CategoricalDtype must be passed to perform an update, '
                    'got {dtype!r}').format(dtype=dtype)
             raise ValueError(msg)
-        elif dtype.categories is not None and dtype.ordered is self.ordered:
-            return dtype
 
         # dtype is CDT: keep current categories/ordered if None
         new_categories = dtype.categories
         if new_categories is None:
             new_categories = self.categories
 
-        new_ordered = dtype.ordered
+        new_ordered = dtype._ordered
+        new_ordered_from_sentinel = dtype._ordered_from_sentinel
         if new_ordered is None:
-            new_ordered = self.ordered
+            # maintain existing ordered if new dtype has ordered=None
+            new_ordered = self._ordered
+            if self._ordered and new_ordered_from_sentinel:
+                # only warn if we'd actually change the existing behavior
+                msg = ("Constructing a CategoricalDtype without specifying "
+                       "`ordered` will default to `ordered=False` in a future "
+                       "version; `ordered=True` must be explicitly passed in "
+                       "order to be retained")
```

Comment:
Should we explain what the use case is here? (maybe we don't really want to encourage using it ..)

Reply:
added some more detail to the message, but agree that we don't really want to encourage using `ordered=None` so I didn't specifically mention it